packages/url/src/index.js

```diff
@@ -244,9 +252,9 @@ export function isValidFragment( fragment ) {
  * includes query arguments, the arguments are merged with (and take precedent
  * over) the existing set.
  *
- * @param {?string} url  URL to which arguments should be appended. If omitted,
- *                       only the resulting querystring is returned.
- * @param {Object}  args Query arguments to apply to URL.
+ * @param {string} [url='']  URL to which arguments should be appended. If omitted,
```

Comment:
The default value seems like a duplication for me, can't it be infered?

Reply:
This is mostly important for documentation generation. Otherwise, I'm mainly just following the WordPress JSDoc standards posted [here](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#functions).

To answer your question: Yes, the TypeScript compiler can infer this.